src/Columns/ColumnTuple.cpp

```diff
@@ -246,12 +301,31 @@ ColumnPtr ColumnTuple::filter(const Filter & filt, ssize_t result_size_hint) con
 
 void ColumnTuple::expand(const Filter & mask, bool inverted)
 {
+    if (columns.empty())
+    {
+        size_t bytes = countBytesInFilter(mask);
+        if (inverted)
+            bytes = mask.size() - bytes;
+        column_length = bytes;
+        return;
+    }
+
     for (auto & column : columns)
         column->expand(mask, inverted);
+
+    column_length = columns[0]->size();
```

Comment:
If me maintain `column_length` only for 0-size tuples, this is not needed anymore
```suggestion
```